include/cuco/static_map.cuh

```diff
@@ -536,6 +536,47 @@ class static_map {
                                            ValueOut values_out,
                                            cuda_stream_ref stream = {}) const;
 
+  /**
+   * @brief Rebuilds the container.
+   *
+   * @note This function synchronizes the given stream. For asynchronous execution use
+   * `rehash_async`.
+   *
+   * @param stream CUDA stream used for this operation
+   */
+  void rehash(cuda_stream_ref stream = {});
```

Comment:
Why is this overload needed?

Reply:
- For convenience, e.g., if someone just wants to remove any `erased_key_sentinel`s from the container without changing its size.
- It saves a few cycles by not recomputing the extent using `make_window_extent`.
- This is the only overload a container with a static extent can use, since the size is built into the type.

Reply:
> This is the only overload a container with a static extent can use, since the size is built into the type.

I would expect a static check in other `rehash` overloads then

Reply:
I can add a static assert. Right now, calling this function with a statically sized container will still fail at compile time, but the diagnostics aren't that nice.

Reply:
On second consideration, the diagnostics are already sufficient: When we pass the new extent, we force it to be of type `extent_type` aka the same extent type of the container. If the extent isn't convertible, we get a simple type mismatch error which is pretty readable.

Adding a static assert would mean I would need to add a new tparam for the extent argument, and then check if the type passed is valid. That's redundant I think.

Reply:
yeah, that's fine.